hack/verify-crdgen.sh

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env bash
+
+# Copyright 2021 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+
+SCRIPT_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
+echo "SCRIPT_ROOT: ${SCRIPT_ROOT}"
+source "${SCRIPT_ROOT}/hack/lib/init.sh"
+
+kube::golang::verify_go_version
+
+#CRD_OPTIONS="crd:trivialVersions=true,preserveUnknownFields=false"
+CRD_OPTIONS="crd:trivialVersions=true,preserveUnknownFields=false"
+
+# Create a temp directory
+_tmpdir="$(mktemp -d "/tmp/sched-plugins.XXXX")"
+trap "rm -rf ${_tmpdir}" EXIT
+
+# Download controller-gen locally
+CONTROLLER_GEN="${GOPATH}/bin/controller-gen"
+$(go install sigs.k8s.io/controller-tools/cmd/controller-gen@latest)
+
+# Generate CRD
+api_paths="./pkg/apis/scheduling/v1alpha1/...;./vendor/github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/apis/..."
+
+${CONTROLLER_GEN} ${CRD_OPTIONS} paths="${api_paths}" output:crd:artifacts:config=${_tmpdir}/crd/bases
+
+# compare crds individualy
+crd_files="$(ls ./manifests/*/crd.yaml)"
+
+for file in ${crd_files}; do
```

Comment:
It'd be easier if we do this way:

1. Soft-link the *crd.yaml to a specific folder `./manifests/crds/`, and name them with the exact name that `controller-gen` would generate.
    ```
    $ ll manifests/crds
    total 0
    lrwxr-xr-x 1 wei.huang1 30 Sep  3 14:05 scheduling.sigs.k8s.io_elasticquotas.yaml -> ../capacityscheduling/crd.yaml
    lrwxr-xr-x 1 wei.huang1 24 Sep  3 14:05 scheduling.sigs.k8s.io_podgroups.yaml -> ../coscheduling/crd.yaml
    lrwxr-xr-x 1 wei.huang1 32 Sep  3 14:06 topology.node.k8s.io_noderesourcetopologies.yaml -> ../noderesourcetopology/crd.yaml
    ```
2. Just specify the output folder to `./manifests/crds`, so the generate output will overwrite.
3. Then use `git diff` to verify if there are any significant incompatible diffs.
4. In this PR, check in the generated crds, but keeps the original manual-maintained fields:
    - annotations.api-approved.kubernetes.io
    - spec.names.shortNames

I verified locally, which should work.